src/test/java/org/kaazing/gateway/client/impl/autobahn/closehandling/CloseHandlingIT.java

```diff
@@ -21,24 +21,33 @@
 
 package org.kaazing.gateway.client.impl.autobahn.closehandling;
 
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
 import java.net.URI;
 
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
-
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
 import org.kaazing.net.ws.WebSocket;
 import org.kaazing.net.ws.WebSocketFactory;
 import org.kaazing.net.ws.WebSocketMessageReader;
 import org.kaazing.net.ws.WebSocketMessageWriter;
-import org.kaazing.robot.junit.annotation.Robotic;
-import org.kaazing.robot.junit.rules.RobotRule;
 
 public class CloseHandlingIT {
-    @Rule
-    public RobotRule robot = new RobotRule();
+	private final K3poRule k3po = new K3poRule();
+
+	private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
+
+	@Rule
+	public final TestRule chain = outerRule(k3po).around(timeout);
 
-    @Robotic(script = "send256KMessageFollowedByCloseThenPing")
+    @Specification("send256KMessageFollowedByCloseThenPing")
     @Test(timeout = 10000)
```

Comment:
Since you have `Timeout` rule, the timeout parameter should be removed from each individual `@Test` annotation, right?

There seems to be some variation in the timeout value across tests, so remember to check that there's no change in behavior when you remove timeout from the `@Test` annotations.